### PR TITLE
resolve: Avoid emitting an unhelpful cascading resolution error

### LINKED ISSUE #35142: Bad cascading "is not a trait" error
## Meta

```
$ rustc -V
rustc 1.12.0-dev (5556554e2 2016-07-31)
```

## STR

```Rust
use std::Xxx;
struct Foo;
impl Xxx for Foo {}
fn main() {}
```

## Expected result

Single unresolved import error

## Actual result

```
error[E0432]: unresolved import `std::Xxx`. There is no `Xxx` in `std`
 --> <anon>:1:5
  |
1 | use std::Xxx;
  |     ^^^^^^^^

error[E0404]: `Xxx` is not a trait
 --> <anon>:3:6
  |
3 | impl Xxx for Foo {}
  |      ^^^

error: cannot continue compilation due to previous error
```

The E0404 error is a duplicate (BTW, E0404 and E0432 should be switched ;-)).

cc @jseyfried

### PR DESCRIPTION
**Pull request by jseyfried**
Fixes #35142.

### CODE REVIEW

**Comment by arielb1** [comment]
r+ with a `match` instead of an if-then-else.

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -1810,6 +1810,8 @@ impl<'a> Resolver<'a> {
             if let Def::Trait(_) = path_res.base_def {
                 debug!("(resolving trait) found trait def: {:?}", path_res);
                 Ok(path_res)
+            } else if path_res.base_def == Def::Err {
```
Comment (arielb1) [inline src/librustc_resolve/lib.rs]: nit: use the match

**Comment by jseyfried** [comment]
@arielb1 Done.

**Comment by arielb1** [comment]
actual failure
```
command: PATH="C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\stage2\bin;C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\stage2-tools\i686-pc-windows-gnu\release\deps;C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\stage2-rustc\i686-pc-windows-gnu\release\deps;C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\stage2-test\i686-pc-windows-gnu\release\deps;C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\stage2-std\i686-pc-windows-gnu\release\deps;C:\mingw-w64\i686-4.9.1-win32-dwarf-rt_v3-rev1\mingw32\bin;C:\Python27;C:\msys64\mingw32\bin;C:\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\msys64\usr\bin;C:\python27;C:\python27\scripts;C:\program files (x86)\inno setup 5;C:\program files (x86)\CMake\bin;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit;C:\Program Files\Microsoft SQL Server\110\Tools\Binn;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0;C:\Program Files\CMake\bin" C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\stage2\bin\rustc.exe C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\src/test\ui\codemap_tests\two_files.rs -L C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\test\ui --target=i686-pc-windows-gnu -L C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\test\ui\codemap_tests\two_files.stage2-i686-pc-windows-gnu.ui.libaux -C prefer-dynamic -o C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\test\ui\codemap_tests\two_files.stage2-i686-pc-windows-gnu.exe -Crpath -O -Lnative=C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\obj\build\i686-pc-windows-gnu\rust-test-helpers
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
error[E0404]: `Bar` is not a trait
  --> C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\src/test\ui\codemap_tests\two_files.rs:16:6
   |
16 | impl Bar for Baz { }
   |      ^^^
   |
   = note: type aliases cannot be used for traits

error: cannot continue compilation due to previous error


------------------------------------------

thread '[ui] ui\codemap_tests\two_files.rs' panicked at 'explicit panic', C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\src\tools\compiletest\src\runtest.rs:2353
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    [ui] ui\codemap_tests\two_files.rs

test result: FAILED. 10 passed; 1 failed; 1 ignored; 0 measured
```

**Pull request merged**